fil-proofs-tooling/src/bin/benchy/merkleproofs.rs

```diff
@@ -217,19 +228,19 @@ pub fn run(size: usize, proofs_count: usize, validate: bool) -> Result<()> {
 
     match size as u64 {
         // 2 KIB sectors are composed of a single 2KIB tree of arity 8.
-        SECTOR_SIZE_2_KIB => run_tree_bench::<_, U8, U1>(&mut rng, nodes, proofs_count, validate),
+        SECTOR_SIZE_2_KIB => run_base_tree_bench::<_, U8>(&mut rng, nodes, proofs_count, validate),
         // 4 KIB sectors are composed of 2 2KIB trees each of arity 8.
-        SECTOR_SIZE_4_KIB => run_tree_bench::<_, U8, U2>(&mut rng, nodes, proofs_count, validate),
+        SECTOR_SIZE_4_KIB => run_sub_tree_bench::<_, U8, U2>(&mut rng, nodes, proofs_count, validate),
         // 8 MIB sectors are composed of a single 8MIB tree of arity 8.
-        SECTOR_SIZE_8_MIB => run_tree_bench::<_, U8, U1>(&mut rng, nodes, proofs_count, validate),
+        SECTOR_SIZE_8_MIB => run_base_tree_bench::<_, U8>(&mut rng, nodes, proofs_count, validate),
         // 16 MIB sectors are composed of 2 8MIB trees each of arity 8.
-        SECTOR_SIZE_16_MIB => run_tree_bench::<_, U8, U2>(&mut rng, nodes, proofs_count, validate),
+        SECTOR_SIZE_16_MIB => run_sub_tree_bench::<_, U8, U2>(&mut rng, nodes, proofs_count, validate),
         // 512 MIB sectors are composed of a single 512MIB tree of arity 8.
-        SECTOR_SIZE_512_MIB => run_tree_bench::<_, U8, U1>(&mut rng, nodes, proofs_count, validate),
+        SECTOR_SIZE_512_MIB => run_base_tree_bench::<_, U8>(&mut rng, nodes, proofs_count, validate),
         // 1 GIB sectors are composed of 2 512MB trees each of arity 8.
-        SECTOR_SIZE_1_GIB => run_tree_bench::<_, U8, U2>(&mut rng, nodes, proofs_count, validate),
+        SECTOR_SIZE_1_GIB => run_sub_tree_bench::<_, U8, U2>(&mut rng, nodes, proofs_count, validate),
         // 32 GIB sectors are composed of 8 4GIB trees each of arity 8.
-        SECTOR_SIZE_32_GIB => run_tree_bench::<_, U8, U8>(&mut rng, nodes, proofs_count, validate),
+        SECTOR_SIZE_32_GIB => run_sub_tree_bench::<_, U8, U8>(&mut rng, nodes, proofs_count, validate),
         // 64 GIB sectors are composed of 2 32 GIB trees.
```

Comment:
Is `with_shape!` no good here?

Reply:
The way this bench is written, not really.  But, as a separate PR, this bench should be re-written (it can be simplified quite a bit now that we have some more advanced primitives)!